lib/common/xml.c

```diff
@@ -3009,6 +3009,21 @@ pcmk__xe_set_props(xmlNodePtr node, ...)
     va_end(pairs);
 }
 
+void
+pcmk__xe_foreach_child(xmlNode *xml, const char *child_element_name,
+                       void (*handler)(xmlNode *xml, void *userdata),
+                       void *userdata)
+{
+    CRM_ASSERT(xml != NULL && handler != NULL);
+
+    for (xmlNode *node = pcmk__xe_first_child(xml); node != NULL;
+         node = pcmk__xe_next(node)) {
```

Comment:
For efficiency's sake I wouldn't use the convenience functions here, it's simple enough to add (node->type == XML_ELEMENT_NODE) in the if

Reply:
Fixed.